internal/elasticsearch/common.go

```diff
@@ -637,14 +637,49 @@ func newVolumeSource(clusterName, nodeName, namespace string, node api.Elasticse
 		StorageClassName: specVol.StorageClassName,
 	}
 
-	err := persistentvolume.CreateOrUpdatePVC(context.TODO(), client, pvc, persistentvolume.LabelsEqual, persistentvolume.MutateLabelsOnly)
+	// PersistentVolumeClaims can run into ResourceQuota error. Since the create
+	// method will return a 403 when it runs into this issue, the existing pvc
+	// should be found and updated rather than attempt to create it and catch the
+	// already exists error.
+	status, err := doesPersistentVolumeClaimExist(claimName, namespace, client)
+
 	if err != nil {
-		log.Error(err, "Unable to create PersistentVolumeClaim")
+		log.Error(err, "Unable to get PersistentVolumeClaim")
+	}
+
+	if status {
+		err = persistentvolume.UpdatePVC(context.TODO(), client, pvc, persistentvolume.LabelsEqual, persistentvolume.MutateLabelsOnly)
+		if err != nil {
+			log.Error(err, "Unable to update PersistentVolumeClaim")
+		}
+	} else {
+		err = persistentvolume.CreateOrUpdatePVC(context.TODO(), client, pvc, persistentvolume.LabelsEqual, persistentvolume.MutateLabelsOnly)
+		if err != nil {
+			log.Error(err, "Unable to create PersistentVolumeClaim")
+		}
 	}
 
 	return volSource
 }
 
+func doesPersistentVolumeClaimExist(claimName, namespace string, c client.Client) (bool, error) {
```

Comment:
Why is it important to return a bool and the error?  Can we not return only a bool and log the error if one occurs?